packages/@react-stately/virtualizer/src/Virtualizer.ts

```diff
@@ -652,6 +652,12 @@ export class Virtualizer<T extends object, V, W> {
     for (let persistKey of this._persistedKeys) {
       let layoutInfo = this.layout.getLayoutInfo(persistKey);
 
+      // Assuming that any item with a parent is a subitem and we need to
+      // persist the parent layout, unless the parent is "body" a Table special key
+      if (layoutInfo && layoutInfo.parentKey && layoutInfo.parentKey !== 'body') {
```

Comment:
so technically when we get to trees, this may become very nested...

I think we should persist the key and each parent all the way up

my guess would be that in the case of key === 'body' it's already in the map, so it should be safe to add them here as well, you won't get duplicates in the map if the key already exists

Reply:
recursion!